Firestore/Example/FuzzTests/FSTFuzzTestsPrincipal.mm

```diff
@@ -29,15 +29,17 @@
 // Fuzz-test the deserialization process in Firestore. The Serializer reads raw
 // bytes and converts them to a model object.
 void FuzzTestDeserialization(const uint8_t *data, size_t size) {
-  DatabaseId database_id{"project", DatabaseId::kDefault};
-  Serializer serializer{database_id};
-
-  @try {
-    serializer.DecodeFieldValue(data, size);
-  } @catch (...) {
-    // Caught exceptions are ignored because the input might be malformed and
-    // the deserialization might throw an error as intended. Fuzzing focuses on
-    // runtime errors that are detected by the sanitizers.
+  // The Serializer object is static to avoid initializing it every method call.
+  static Serializer serializer{DatabaseId{"project", DatabaseId::kDefault}};
```

Comment:
`Serializer` only contains a single reference, and it's trivially destructible. To be future-proof, a check could optionally be added:
```cpp
#include <type_traits>
// ...
// inside the function
static_assert(std::is_trivially_destructible<Serializer>::value,
    "Serializer shouldn't be static now that it's no longer trivially destructible (see style guide)");
```
Though I think it's a bit overkill.

Reply:
> Serializer only contains a single reference, and it's trivially destructible

Ah, cool. I had just assumed it wasn't. (You'd think I would've known. :)

> Since both the Serializer and DatabaseId objects have low overheads, I am removing the static modifier until I have numbers that show the advantage of having a static object.

This is probably the right call. Premature optimization and all that.